src/app/report/difference-modal/difference-modal.component.ts

```diff
@@ -0,0 +1,54 @@
+import { Component, Input, Output, TemplateRef, ViewChild } from '@angular/core';
+import { ReportDifference } from '../../shared/interfaces/report-difference';
+import { TitleCasePipe } from '@angular/common';
+import { Subject } from 'rxjs';
+import { NgbModal, NgbModalRef } from '@ng-bootstrap/ng-bootstrap';
+
+export const changesActionConst = ['save', 'discard'] as const;
+export type ChangesAction = (typeof changesActionConst)[number];
+
+@Component({
+  selector: 'app-difference-modal',
+  standalone: true,
+  imports: [TitleCasePipe],
+  templateUrl: './difference-modal.component.html',
+  styleUrl: './difference-modal.component.css',
+})
+export class DifferenceModalComponent {
+  protected saveOrDiscardType!: ChangesAction;
+  protected reportDifferences?: ReportDifference[];
+  protected activeModal?: NgbModalRef | undefined;
```

Comment:
```suggestion
  protected activeModal?: NgbModalRef;
```
The `?` already marks it as possibly undefined :)

Reply:
fixed